extensions/amp-viewer-integration/0.1/test/test-amp-viewer-integration.js

```diff
@@ -14,9 +14,193 @@
  * limitations under the License.
  */
 
+import {Messaging} from '../messaging.js';
+import {Viewer} from './viewer-messaging-example.js';
+import {adopt} from '../../../../src/runtime';
+import * as sinon from 'sinon';
+
+
+adopt(window);
 
 describes.sandboxed('AmpViewerIntegration', {}, () => {
-  it('should work', () => {
-    // TODO
+  const ampDocSrc = '/test/fixtures/served/ampdoc-with-messaging.html';
+
+  let viewerEl;
+  let viewer;
+
+  beforeEach(() => {
+    const loc = window.location;
+    const ampDocUrl =
+      `${loc.protocol}//iframe.${loc.hostname}:${loc.port}${ampDocSrc}`;
+
+    viewerEl = document.createElement('div');
+    document.body.appendChild(viewerEl);
+    viewer = new Viewer(viewerEl, '1', ampDocUrl, true);
+    return viewer.waitForHandshakeRequest();
+  });
+
+  afterEach(() => {
+    document.body.removeChild(viewerEl);
+  });
+
+  describe('Handshake', function() {
```

Comment:
This `describe` seems to be the only 1 using the `beforeEach` and `afterEach` above. Could this be separated from the other tests?